### PR TITLE
[LTC] Upstream helpers for c10::Device <=> BackendDevice

### PR DESCRIPTION
**Pull request by alanwaketan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #69064

Summary:
This commit upstreams helpers for converting a c10::Device to
BackendDevice and vice versa.

Test Plan:
./build/bin/test_lazy --gtest_filter=BackendDeviceTest.FromAten:BackendDeviceTest.ToAten

Differential Revision: [D32732607](https://our.internmc.facebook.com/intern/diff/D32732607)

### CODE REVIEW

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32732607).

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32732607).

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32732607).

**Review by wconstab — APPROVED** [review]

**Comment by alanwaketan** [comment]
Thanks for approving the pull request, Will.

**Review by JackCaoG — APPROVED** [review]
Thanks

File: torch/csrc/lazy/backend/backend_device.cpp [hunk 1]
```diff
@@ -11,7 +13,7 @@ namespace lazy {
 BackendDevice::BackendDevice()
   : type_(std::make_shared<BackendDeviceType>()) {}
 
-BackendDevice::BackendDevice(std::shared_ptr<BackendDeviceType>&& type, int ordinal)
+BackendDevice::BackendDevice(std::shared_ptr<BackendDeviceType>&& type, int64_t ordinal)
```
Comment (desertfire) [inline torch/csrc/lazy/backend/backend_device.cpp:16]: Nit: any specific reason we want to move to int64 here?

**Review by desertfire — APPROVED** [review]

File: torch/csrc/lazy/backend/backend_device.cpp (hunk 1 above)
Comment (alanwaketan) [reply torch/csrc/lazy/backend/backend_device.cpp:16]: It's usually better to specify the length of the int. Usually, just use int64_t given we are now in the age of 64 bit.

**Pull request closed without merging**